tap_trello/streams.py

```diff
@@ -154,24 +138,24 @@ def _paginate_window(self, window_start, window_end, format_values, params):
             if len(records) >= MAX_API_RESPONSE_SIZE:
                 LOGGER.info("%s - Paginating within date_window %s to %s, due to max records being received.",
                             self.stream_id,
-                            utils.strftime(window_start), utils.strftime(window_end))
+                            utils.strftime(window_start), utils.strftime(sub_window_end))
                 # NB: Actions are sorted backwards, so if we get the
                 # max_response_size, set the window_end to the last
                 # record's timestamp (inclusive) and try again.
-                window_end = utils.strptime_to_utc(records[-1]["date"]) + timedelta(milliseconds=1)
-                self._update_bookmark("window_sub_end", window_end)
+                sub_window_end = utils.strptime_to_utc(records[-1]["date"]) + timedelta(milliseconds=1)
+                self._update_bookmark("sub_window_end", sub_window_end)
                 singer.write_state(self.state)
             else:
                 LOGGER.info("%s - Finished syncing between %s and %s",
                             self.stream_id,
                             utils.strftime(window_start),
-                            singer.get_bookmark(self.state, self.stream_id, "next_window_start"))
-                singer.bookmarks.clear_bookmark(self.state, self.stream_id, "window_start")
-                singer.bookmarks.clear_bookmark(self.state, self.stream_id, "window_sub_end")
-                singer.bookmarks.clear_bookmark(self.state, self.stream_id, "next_window_start")
+                            window_end)
+                singer.bookmarks.clear_bookmark(self.state, self.stream_id, "sub_window_end")
+                singer.write_state(self.state)
```

Comment:
This state looks like the initial state before paginating, so it might be good to hold off and let the parent_id logic to write the next state.